src/main/scala/is/hail/annotations/aggregators/RegionValueCollectAggregator.scala

```diff
@@ -62,3 +63,38 @@ class RegionValueCollectIntAggregator extends RegionValueAggregator {
     ab.clear()
   }
 }
+
+class RegionValueCollectBaseStructAggregator() extends RegionValueAggregator {
+  private val ab = new MissingBaseStructArrayBuilder()
+  private lazy val t = inputTyp.asInstanceOf[TBaseStruct]
+
+  def seqOp(region: Region, offset: Long, missing: Boolean) {
+    if (missing)
+      ab.addMissing()
+    else
+      ab.add(SafeRow(t, region, offset))
```

Comment:
I suspect its faster to use an `Encoder` rather than convert to annotation.

Reply:
Can you give an example? The Encoder classes I saw all involved Code.